src/extension.ts

```diff
@@ -90,24 +123,28 @@ export function activate(context: vscode.ExtensionContext) {
         return positions;
     }
 
-    function readConfiguration(): { mode: string, regex: RegExp } {
-        const ligatureConfiguration = vscode.workspace.getConfiguration().get("disableLigatures") as any;
-        const ligatures = ligatureConfiguration.ligatures as string[];
+    function readConfiguration(): { mode: string, regex: RegExp, globalRegex: RegExp } {
+        const ligatureConfiguration = Object(vscode.workspace.getConfiguration().get("disableLigatures")) as any;
+        const globalLigatures = Array.from(ligatureConfiguration.globalLigatures || []) as string[];
+        const ligatures = Array.from(ligatureConfiguration.ligatures || []) as string[];
+        const sortByLongestFirst = (a: any, b: any) => b.length - a.length;
 
         // Match longest first
-        ligatures.sort((a, b) => {
-            return b.length - a.length;
-        });
+        globalLigatures.sort(sortByLongestFirst);
+        ligatures.sort(sortByLongestFirst);
 
         const escapeRegex = /[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g;
 
         // tslint:disable-next-line:prefer-for-of
         for (let i = 0; i < ligatures.length; i++) {
+            globalLigatures[i] = globalLigatures[i].replace(escapeRegex, "\\$&");
             ligatures[i] = ligatures[i].replace(escapeRegex, "\\$&");
         }
 
-        const mode = ligatureConfiguration.mode as string;
-        const regex = new RegExp(ligatures.join("|"), "gi");
-        return { mode, regex };
+        const mode = (ligatureConfiguration.mode || '') as string;
+        const globalHashMap = globalLigatures.reduce((hashMap, string) => Object.assign(hashMap, { [string]: true }), {});
+        const globalRegex = new RegExp(globalLigatures.join("|") || "[^\\W\\w]", "gi");
```

Comment:
What does the fallback regex do? Would it make sense if nothing is defined to not do any regex?

Reply:
Also curious about what the defaulted regex does. I would think that it should just be `const globalRegex = new RegExp(ligatures.join("|"), "gi");`